Client/DataAccess/SQL/HistoryTable.swift

```diff
@@ -5,111 +5,59 @@
-        ])
-        return 0
+        return ("DELETE FROM \(TableNameHistory)", args)
     }
 
-    func update<T>(db: SQLiteDBConnection, item: T?, inout err: NSError?) -> Int {
-        debug("Update into \(name) \(item)")
-        if let site = item as? Site {
-            let query = "UPDATE \(self.name) SET title = ? WHERE guid = ? AND url = ?"
-            let args = [site.title, site.guid, site.url]
-            if let error = db.executeChange(query, withArgs: args) {
-                println(error.description)
-                err = error
-                return 0
-            }
-
-            return db.numberOfRowsModified
+    override var factory: ((row: SDRow) -> Site)? {
+        return { row -> Site in
+            let site = Site(url: row[1] as String, title: row[2] as String)
+            site.guid = row[0] as? String
+            return site
         }
-
-        err = NSError(domain: "mozilla.org", code: NotASiteErrorCode, userInfo: [
-            NSLocalizedDescriptionKey: "Tried to save something that isn't a site"
-        ])
-        return 0
     }
 
-    func delete<T>(db: SQLiteDBConnection, item: T?, inout err: NSError?) -> Int {
-        debug("Delete from \(name) \(item)")
-        var numDeleted: Int = 0
-        var str = "DELETE FROM \(name)"
-        var args = [String]()
-
-        if let site = item as? Site {
-            str += " WHERE url = ?"
-            args.append(site.url)
-        } else if item != nil {
-            err = NSError(domain: "org.mozilla", code: 1, userInfo: [NSLocalizedDescriptionKey: "Invalid object"])
-            return numDeleted
+    override func getQueryAndArgs(options: QueryOptions?) -> (String?, [AnyObject?]) {
+        var args = [AnyObject?]()
+        if let filter = options?.filter {
+            args.append(filter)
+            return ("SELECT guid, url, title FROM \(TableNameHistory) WHERE url LIKE ?", args)
         }
-
-        if str != "" {
-            if let error = db.executeChange(str, withArgs: args) {
-                println(error.description)
-                err = error
-                return 0
-            }
-        }
-
-        return db.numberOfRowsModified
-    }
-
-    private func fromResult(result: SDRow) -> Site {
-        let site = Site(url: result[1] as String, title: result[2] as String)
-        site.guid = result[0] as String
-        return site
-    }
-
-    func factory(row: SDRow) -> Site {
-        let url = row["url"] as String
-        let title = row["title"] as String
-        let guid = row["guid"] as String
-
-        let s = Site(url: url, title: title)
-        s.guid = guid
-        return s
-    }
-
-    func query(db: SQLiteDBConnection) -> Cursor {
-        debug("Query \(name)")
-        return db.executeQuery("SELECT guid, url, title FROM \(name)", factory: self.factory)
+        return ("SELECT guid, url, title FROM \(TableNameHistory)", args)
```

Comment:
The way this is written now, querying the HistoryTable would do that. Querying the JointHistoryVisitsTable would (I think) not (I'll add a test for that!). I've been using QueryOptions as a way to change that behavior if you wanted. i.e. to get all the History entries with zero visits, you'd query the JointHistoryVisitsTable with some option set.